### PR TITLE
fix: add end-of-options separator before branch names in git commands

### PR DESCRIPTION
**Pull request by doomspork**
## Summary

Adds `--` (POSIX end-of-options separator) before user-provided ref names in git command invocations across `src/git.rs` and `src/diff.rs`. This prevents ref arguments from being misinterpreted as git options.

**5 call sites updated:**
- `branch_delete` — both `-d` and `-D` paths
- `restore_worktree` — `worktree add` with existing branch
- `rename_branch` — `branch -m`
- `merge_base` — in `diff.rs`

**Call sites confirmed safe (unchanged):**
- `create_worktree` — `-b` flag consumes the next arg
- `has_unmerged_commits` / range format args — `base` from `default_branch()` is always safe
- `file_diff` / `revert_file` — already use `--`; merge_base result is a hex SHA

## Complexity Notes

While git's `check-ref-format` rules forbid branch names starting with `-`, the `--` separator is zero-cost defense-in-depth that protects against any edge cases with arbitrary ref names. Functions like `merge_base` accept arbitrary refs, not just branch names.

## Test Steps

1. Run `cargo test --all-features` — all 191 tests pass
2. Run `cargo clippy --workspace --all-targets` — clean
3. Run `cargo fmt --all --check` — clean
4. Verify existing branch operations (create workspace, delete workspace, rename workspace) work as before

## Checklist

- [x] Tests added/updated
- [x] Documentation updated (if applicable)

### CODE REVIEW

File: src/git.rs [hunk 1]
```diff
@@ -161,7 +161,11 @@ pub async fn restore_worktree(
     branch_name: &str,
     worktree_path: &str,
 ) -> Result<String, GitError> {
-    run_git(repo_path, &["worktree", "add", worktree_path, branch_name]).await?;
+    run_git(
+        repo_path,
+        &["worktree", "add", worktree_path, "--", branch_name],
+    )
+    .await?;
```
[the remark this answers is not in the document]
Comment (doomspork) [reply src/git.rs:168]: Done — added test_restore_worktree which creates a worktree, removes it, then calls restore_worktree to re-attach it and asserts the branch and cleanup work correctly.
